src/csharp/.editorconfig

```diff
@@ -1,5 +1,6 @@
 root = true
 [**]
+charset = utf-8  ; Implies no BOM.
```

Comment:
Can we refrain from enhancements like this in an otherwise huge PR? Like this I have no idea if the changes is actually necessary for things to work or if it's just a nit you saw and decided to fix.
I'll be happy to accept improvements like this in a separate PR, that would be much easier to review and trivial to check for error.

Reply:
Sure. I'll leave only strictly the main Grpc.Tools changes is this branch then.